common/src/main/java/org/candlepin/common/exceptions/mappers/BadQueryParameterHandler.java

```diff
@@ -0,0 +1,97 @@
+package org.candlepin.common.exceptions.mappers;
+
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.inject.Provider;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.Response.ResponseBuilder;
+import javax.ws.rs.core.Response.Status;
+
+import org.apache.commons.lang.StringUtils;
+import org.candlepin.common.exceptions.CandlepinParameterParseException;
+import org.candlepin.common.exceptions.ExceptionMessage;
+import org.candlepin.common.util.VersionUtil;
+import org.xnap.commons.i18n.I18n;
+
+import com.google.inject.Inject;
+
+/**
+ * Class is able to detect whether an exception contains error related to wrong
+ * parameters. In addition to that this class is able to build an http error response
+ * for that exception.
+ * 
+ * Historical notes: This had to be extracted from BadRequestExceptionMapper, because the new
+ * RestEasy changed the way it handles (now throws NotFoundException)
+ * unparseable parameters of JAX-RS resources.
+ * 
+ * @author fnguyen
+ *
+ */
+public class BadQueryParameterHandler {//extends CandlepinExceptionMapper {
+	private static final Pattern PARAM_REGEX = Pattern.compile("(?:javax\\.ws\\.rs\\.\\w+\\(\\\")([\\w\\s]+)(\\\"\\))");
+	private static final Pattern ILLEGAL_VAL_REGEX = Pattern.compile(":?value\\sis\\s'([\\w\\s]+)(:?'\\sfor)");
+
+
+    @Inject
+    private Provider<I18n> i18n;
+    
+    @Inject 
+    private CandlepinExceptionMapper cem;
+    
+	/**
+	 * @param exception
+	 * @return True if and only if this class can handle the exception.
+	 */
+	public boolean canHandle(Exception exception) {
+		Throwable cause = exception.getCause();
+
+		if (cause instanceof CandlepinParameterParseException) {
+			return true;
+		}
+		String msg = exception.getMessage();
+
+		if (StringUtils.isNotEmpty(msg)) {
+			Matcher paramMatcher = PARAM_REGEX.matcher(msg);
+			Matcher illegalValMatcher = ILLEGAL_VAL_REGEX.matcher(msg);
+			if (paramMatcher.find() && illegalValMatcher.find()) {
+				if ((paramMatcher.groupCount() & illegalValMatcher.groupCount()) == 2) {
+					return true;
+				}
+			}
+		}
+
+		return false;
+	}
+
+	/**
+	 * Builds response for this exception. 
+	 * 
+	 * @param exception
+	 * @return
```

Comment:
This javadoc is missing information about the details of what this method does, the expected parameter, and what it returns.